docs/tutorials/typescript/nextjs/index.md

```diff
@@ -1,290 +1,626 @@
+Now create a directory to hold your Temporal Workflows and Activities:
 
-```js
-// /temporal/tsconfig.json
+```command
+mkdir -p temporal/src
+```
+
+Configure TypeScript to compile from `temporal/src` to `temporal/lib` by adding a new `tsconfig.json` in the `temporal/src` folder:
+
+```command
+touch temporal/tsconfig.json
+```
+
+Add the following configuration to the file:
+
+```json
 {
-  "extends": "@tsconfig/node16/tsconfig.json", // optional but nice to have
+  "extends": "@tsconfig/node20/tsconfig.json",
   "version": "4.4.2",
   "compilerOptions": {
-    "emitDecoratorMetadata": false,
-    "experimentalDecorators": false,
     "declaration": true,
     "declarationMap": true,
     "sourceMap": true,
-    "composite": true,
     "rootDir": "./src",
     "outDir": "./lib"
   },
-  "include": ["src/**/*.ts"],
-  "exclude": ["node_modules"]
+  "include": ["src/**/*.ts"]
 }
 ```
 
-For convenience, you may want to set up some npm scripts to run the builds in your project root `package.json`.
+For convenience, set up some scripts to run the builds in your project root's `package.json`.
 
+Add the `npm-run-all` command to your project as a dependency:
 
-```js
-// /package.json
+```command
+npm install npm-run-all --save-dev
+```
+
+Then use `npm-run-all` to define scripts to run your Temporal build processes and run the Next.JS app alongside Temporal Workers.
+
+Locate your existing `scripts` section, which defines  scripts to run and build Next.js applications:
+
+```json
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+```
+
+Change the scripts so you can build and run your Next.js application and your Temporal Workers at the same time:
+
+```json
   "scripts": {
     "dev": "npm-run-all -l build:temporal --parallel dev:temporal dev:next start:worker",
     "dev:next": "next dev",
     "dev:temporal": "tsc --build --watch ./temporal/tsconfig.json",
     "build:next": "next build",
     "build:temporal": "tsc --build ./temporal/tsconfig.json",
-    "start": "npm run dev",
     "start:worker": "nodemon ./temporal/lib/worker",
+    "start": "next start",
     "lint": "eslint ."
   },
 ```
 
-In the above example we use `npm-run-all` and `nodemon` so that we are able to do 4 things:
+These scripts let you run the following steps with a single `npm run dev` command:
+
+- build Temporal once.
+- start Next.js locally.
+- start a Temporal Worker.
+- rebuild Temporal files when they change.
+
+With your project configured, you can write your Temporal Workflows and Activities.
 
-- build Temporal once
-- start Next.js locally
-- start a Temporal Worker
-- rebuild Temporal files on change
+## Define the business logic using Temporal
 
-in a single `npm run dev` command.
+You'll use Temporal to power a one-click ordering process.
+You'll use a Temporal Workflow to represent each order.
+That Workflow will orchestrates one or more Activities.
```

Comment:
I think it's important to hone in on this a bit especially for Temporal beginners. Maybe something like:

That Workflow will orchestrates one or more Activities, meaning that the Workflow will manage the state and sequence of Activities, which can involve retries, handling failures, and managing timeouts.

Reply:
Reworked to reduce some duplication while keeping the introduction and "signpost" intact.